### PR TITLE
Structured data source driver

### PR DESCRIPTION
**Pull request by puerco**
# Summary

This commit adds the structured data driver to minder. This data source lets us reuse definitions to locate and read data from yaml/json/toml files stored in a repsitory. This is the first version that only supports alternative paths of a file. 

Here's an example data source that loads the security insights file:

```yaml
version: v1
type: data-source
name: securityinsights
context: {}
structured:
  def:
    "parse":
      path:
        file_name: SECURITY-INSIGHTS.yaml
        alternatives:
        - "SECURITY-INSIGHTS.yml"
        - "security-insights.yaml"
        - "security-insights.yml"
        - "SECURITY_INSIGHTS.yaml"
        - "SECURITY_INSIGHTS.yml"

```

## Change Type

***Mark the type of change your PR introduces:***

- [ ] Bug fix (resolves an issue without affecting existing features)
- [x] Feature (adds new functionality without breaking changes)
- [ ] Breaking change (may impact existing functionalities or require documentation updates)
- [ ] Documentation (updates or additions to documentation)
- [ ] Refactoring or test improvements (no bug fixes or new functionality)

# Testing

***Outline how the changes were tested, including steps to reproduce and any relevant configurations. 
Attach screenshots if helpful.***

# Review Checklist:

- [x] Reviewed my own code for quality and clarity.
- [x] Added comments to complex or tricky code sections.
- [ ] Updated any affected documentation.
- [x] Included tests that validate the fix or feature.
- [ ] Checked that related changes are merged.

### CODE REVIEW

File: internal/datasources/service/convert.go [hunk 1]
```diff
@@ -60,3 +62,26 @@ func dataSourceRestDBToProtobuf(ds *minderv1.DataSource, dsfuncs []db.DataSource
...
+	for _, dsf := range dsfuncs {
+		key := dsf.Name
+		dsfToParse := &minderv1.StructDataSource_Def{
+			Path: &minderv1.StructDataSource_Def_Path{},
+		}
+		if err := protojson.Unmarshal(dsf.Definition, dsfToParse); err != nil {
+			return nil, fmt.Errorf("failed to unmarshal data source definition for %s: %w", key, err)
+		}
+
+		ds.GetStructured().Def[key] = dsfToParse
```
Comment (JAORMX) [inline internal/datasources/service/convert.go:83]: this looks panicky. Should we verify that `GetStructured` isn't nil?

Comment (puerco) [reply internal/datasources/service/convert.go:83]: Can't be nil, it's defined on line 69 (it's analogous to the REST equivalent on L60)

Comment (puerco) [reply internal/datasources/service/convert.go]: I refactored the func to make it clearer, PTAL!

File: proto/minder/v1/minder.proto [hunk 2]
```diff
@@ -4036,11 +4036,33 @@ message DataSource {
...
 
     oneof driver {
+        // structured is the structired data - data source.
+        StructDataSource structured = 8;
         // rest is the REST data source driver.
         RestDataSource rest = 6;
     }
 }
 
+// StructDataSource is the REST data source driver.
```
Comment (JAORMX) [inline proto/minder/v1/minder.proto:4046]: You probably want to make this comment relevant to this data source.

File: proto/minder/v1/minder.proto [hunk 3]
```diff
@@ -4036,11 +4036,33 @@ message DataSource {
...
+
+        message Path {
+            string file_name = 1;
+            repeated string alternatives = 2 [
+                (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+            ];
+        }
+    }
+
+    // defs is the list of definitions for the REST API.
```
Comment (JAORMX) [inline proto/minder/v1/minder.proto:4062]: You probably want to make this comment relevant to this data source.

File: proto/minder/v1/minder.proto [hunk 4]
```diff
@@ -4036,11 +4036,33 @@ message DataSource {
...
+        StructDataSource structured = 8;
         // rest is the REST data source driver.
         RestDataSource rest = 6;
     }
 }
 
+// StructDataSource is the REST data source driver.
+message StructDataSource {
+    message Def {
+        // Path is the path specification for the structured data daou
```
Comment (JAORMX) [inline proto/minder/v1/minder.proto:4049]: daou?

**Review by JAORMX — DISMISSED** [review]
The comments I left can be addressed in a subsequent PR.

**Comment by puerco** [comment]
OK, I fixed them now so that they don't show up in the docs, thanks for catching these :+1:

**Review by JAORMX — APPROVED** [review]
shipit

**Pull request merged**